prompb/types.pb.go

```diff
@@ -430,6 +432,20 @@ func (m *ReadHints) GetEndMs() int64 {
 	return 0
 }
 
+func (m *ReadHints) GetGroupBy() []string {
+	if m != nil {
+		return m.GroupBy
+	}
+	return nil
+}
+
+func (m *ReadHints) GetWithout() bool {
+	if m != nil {
+		return m.Without
+	}
+	return false
```

Comment:
If missing this would mean `by()`, which is not the same as not having a grouping. True is the right default here.

Reply:
will correct it

Reply:
Thinking further, we actually want to change the field to By (rather than Without) and default to false. That way a Prometheus that doesn't support this looks the same as one that does support it but wants no grouping.

Reply:
Yes, agree, I did change it to `By` and default set to false, please see https://github.com/prometheus/prometheus/pull/6401/commits/95cb9b4ba4b627a903f044b4c8c8b090e609371b